newton/_src/geometry/narrow_phase.py

```
@@ -74,6 +80,163 @@
+    work_items: wp.array[int],
+    work_count: wp.array[int],
+):
+    """Append one work item using a single atomic reservation per active warp."""
+    mask = _warp_ballot(int(predicate))
+    if predicate:
+        lane = _warp_lane_id()
+        leader = _warp_first_lane(mask)
+        count = _warp_population_count(mask)
+        lower_mask = mask & ((wp.uint32(1) << wp.uint32(lane)) - wp.uint32(1))
+        rank = _warp_population_count(lower_mask)
+        base = int(0)
```

Ruff (0.16.1)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)